firefox-ios/Client/Coordinators/Browser/BrowserCoordinator.swift

```diff
@@ -1057,38 +1130,27 @@ class BrowserCoordinator: BaseCoordinator,
 
     // MARK: - Private helpers
 
-    private func showShareSheet(with url: URL?, title: String?) {
-        guard let url else { return }
-
-        let showShareSheet = { url in
-            self.startShareSheetCoordinator(
-                url: url,
-                title: title,
-                sourceView: self.browserViewController.addressToolbarContainer,
-                sourceRect: nil,
-                toastContainer: self.browserViewController.contentContainer,
-                popoverArrowDirection: .any
-            )
-        }
-
-        // We only care about the temp document for .tab shares right?
-        // BUT!! A tab might be displaying a downloaded PDF! Then it's file:// url
-        guard let temporaryDocument = browserViewController.tabManager.selectedTab?.temporaryDocument else {
-            showShareSheet(url)
-            return
-        }
+    private func tryDownloadingTabFileToShare(shareType: ShareType) async -> ShareType {
```

Comment:
I don't think we should have this download code in a coordinator. Coordinators should only contain navigation logic as much as possible. This makes me think we need a sort of ShareDownloadManager that can handle files download?

Reply:
Yeah this code was originally in 2 different coordinators... at least it's just in 1 now!! 😆  It's really not ideal. The FXIOS-10749 ticket I filed will take care of this, so let's save it for later as it's a bigger change and I need to ask Andy to check with the UI team about changing this functionality.

In my opinion, the current implementation needs some rethinking, because there's no responsiveness in the UI when waiting for large PDF files to download prior to showing the share sheet. Ideally the temporary file is saved when it's downloaded for the webview and then the coordinator would have none of this logic. It would either share the file (if it's downloaded already) or share the file's online URL as a fallback. Safari is doing something like that currently.

Reply:
Ok we could maybe add a small comment in the meantime above this method (that FXIOS-10749 will move this logic elsewhere?). I am scared of our `BrowserCoordinator` becoming the new `BrowserViewController` at some point with a lot of logic in there. Just want to be careful around it!

Reply:
Yes that's a good call! 

I just realized I gave you the wrong ticket, sorry about that. It's actually FXIOS-10824, and I already have a comment for this exact thing on line 819 where `BrowserCoordinator` calls `tryDownloadingTabFileToShare`. Is that ok or would you like me to elaborate on the coordinator situation a bit? 🙂

Reply:
Seems good then thank you!